### PR TITLE
Deprecate PSP-related types in extensions/v1beta1

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it**:
This PR deprecates PSP-related types in `extensions/v1beta1` API Group and suggests to use their versions from `policy/v1beta1`. This is a part of PSP migration away from `extensions` API Group.

**Which issue(s) this PR fixes**:
Addressed to https://github.com/kubernetes/features/issues/5

### CODE REVIEW

**Comment by php-coder** [comment]
/release-note-none
/sig auth

PTAL @kubernetes/sig-auth-api-reviews 
CC @simo5

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 1]
```diff
@@ -864,6 +864,7 @@ type ReplicaSetCondition struct {
 
 // PodSecurityPolicy governs the ability to make requests that affect the Security Context
 // that will be applied to a pod and container.
+// Deprecated: use PodSecurityPolicy from policy/v1beta1 API Group instead.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go]: omit the version... we don't want to need to update this when it gets promoted in the policy group

**Comment by php-coder** [comment]
@liggitt Thanks! Your comment was addressed and it's green. PTAL.

**Pull request merged**